File: MenuLibrary.java
import java.util.Random;

public class MenuLibrary {
    public static void libraryMenu(){//Método menú para la sección Librería
        boolean flag = true;
        int x;
        do{
            System.out.println("\nSelecciona una opción:\n1) Registrar Libro\n2) Registrar Usuario\n3) Libros\n4) Usuarios\n5)Salir");
            System.out.print("\nOpción: ");
            x = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (x){
                case 1 ->{
                    registerBook();
                }
                case 2 ->{
                    registerUser();
                }
                case 3 ->{
                    menuBooks();
                }
                case 4 ->{
                    menuUsers();
                }
                case 5 ->{
                    System.out.println("\nRegresando al menú principal...");
                    flag = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida.");
            }
        }while (flag);
    }
    //Métodos diseñados para cada sección del menú
    public static void registerBook(){
        Random number = new Random();
        boolean flag = true;
        int opc, year, id;
        String title, author;
        do{
            System.out.println("\nConfirmar:\n1) Sí\n2)No");
            System.out.print("\nOpción: ");
            opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    System.out.print("\nIngresa el título: ");
                    title = Reader.sc.nextLine();
                    System.out.print("\nIngresa el autor: ");
                    author = Reader.sc.nextLine();
                    System.out.print("\nIngresa el año de publicación: ");
                    year = Reader.sc.nextInt();
                    Reader.sc.nextLine();
                    System.out.println("\nRegistrando...");
                    id = number.nextInt();
                    Books book = new Books(id,title,author,year);
                    System.out.println("\nAñadiendo a biblioteca...");
                    Library.addBook(book);
                    System.out.println("\nLibro registrado!");
                    flag = false;
                }
                case 2 ->{
                    System.out.println("\nRegresando al menú anterior...\n");
                    flag = false;
                }
                default -> System.out.println("Error. Elige una opción válida.");
            }
        }while(flag);
    }
    public static void registerUser(){
        Random number = new Random();
        boolean sig = true;
        int opc, idUser,cellNumber,age;
        String name, lastName, address;
        do{
            System.out.println("\nConfirmar:\n1) Sí\n2)No");
            System.out.print("\nOpción: ");
            opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    System.out.print("\nIngresa el nombre: ");
                    name = Reader.sc.nextLine();
                    System.out.print("\nIngresa los apellidos: ");
                    lastName = Reader.sc.nextLine();
                    System.out.print("\nIngresa la dirección: ");
                    address = Reader.sc.nextLine();
                    System.out.print("\nIngresa un número de teléfono: ");
                    cellNumber = Reader.sc.nextInt();
                    System.out.print("\nIngresa la edad: ");
                    age = Reader.sc.nextInt();
                    if(age < 18){
                        System.out.println("Error. El usuario es menor de edad.\nRegresando al menú...");
                    }else{
                        System.out.println("\nRegistrando usuario..");
                        idUser = number.nextInt(100000,999999);
                        Users user = new Users(idUser,name,lastName,age,cellNumber,address);
                        System.out.println("\nAñadiendo a base de datos...");
                        Library.addUser(user);
                        System.out.println("\nUsuario registrado!");
                    }
                    sig = false;
                }
                case 2 ->{
                    System.out.println("\nRegresando al menú anterior...\n");
                    sig = false;
                }
                default -> System.out.println("Error. Elige una opción válida.");
            }
        }while(sig);
    }
    public static void menuBooks(){
        boolean pt = true, op = true, st = true;
        int opc,b,j,k,y,r,a;
        do{
            System.out.println("\nSelecciona la opción deseada:\n1) Consultar biblioteca (rentas)\n2) Consultar librería (ventas)\n3) Libros disponibles para renta\n4) Libros disponibles en venta\n5) Eliminar Libro\n6) Salir");
            System.out.print("\nOpción: ");
            opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    for(Books book:Library.getBooksList()){
                        System.out.printf("\nID: %s\nTítulo: %s\nAutor: %s\nAño de Publicación: %d\n",book.getID(),book.getTitle(),book.getAuthor(),book.getYear());
                        System.out.println("**************");
                    }
                }
                case 2 ->{
                    for(SaleBooks book:Library.getBooksSell()){
                        System.out.printf("\nID: %s\nTítulo: %s\nAutor: %s\nAño de Publicación: %d\n",book.getID(),book.getTitle(),book.getAuthor(),book.getYear());
                        System.out.println("**************");
                    }
                }
                case 3 ->{
                    b=0;//Ciclo para contar los libros disponibles.
                    for(Books books:Library.getBooksList()){
                        if(books.isRented()) b+=1;
                    }
                    for(Books book:Library.getBooksList()){
                        if(!book.isRented()){
                            System.out.printf("\nID: %s\nTítulo: %s\nAutor: %s\nAño de Publicación: %d\n",book.getID(),book.getTitle(),book.getAuthor(),book.getYear());
                            System.out.println("**************");
                        }else if(b==0) System.out.println("\nNo hay libros disponibles.");
                    }
                }
                case 4 ->{
                    b = 0;//Ciclo para contar libros disponibles.
                    for(SaleBooks books:Library.getBooksSell()){
                        if(books.isAvailable()) b+=1;
                    }
                    for(SaleBooks book:Library.getBooksSell()){
                        if(book.isAvailable()){
                            System.out.printf("\nID: %s\nTítulo: %s\nAutor: %s\nAño de Publicación: %d\n",book.getID(),book.getTitle(),book.getAuthor(),book.getYear());
                            System.out.println("**************");
                        }else if(b==0) System.out.println("\nNo hay libros disponibles.");
                    }
                }
                case 5 ->{
                    do{
                        System.out.println("\n1) Libros para Renta\n2) Libros para Venta\n3) Salir");
                        a = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        switch (a){
                            case 1 ->{
                                j = 1;
                                System.out.println("\t\nSelecciona el libro a eliminar:\n");
                                for (Books book : Library.getBooksList()){
                                    System.out.printf("\n%d) ID: %s\n   Nombre: %s\n   Autor: %s",j,book.getID(),book.getTitle(),book.getAuthor());
                                    j++;
                                }
                                do {
                                    y = 0;
                                    System.out.print("\nRespuesta: ");
                                    y = Reader.sc.nextInt();
                                    Reader.sc.nextLine();
                                    y -= 1;
                                    if (y > Library.getBooksList().size()) System.out.println("Error. Ingresa una opción válida.");
                                    else op = false;
                                } while (op);
                                Books book = Library.getBooksList().get(y);
                                System.out.println("\nSeguro/a que deseas eliminar este libro?\n1) Sí\n2) No");
                                op = true;
                                do {
                                    r = 0;
                                    System.out.print("\nRespuesta: ");
                                    r = Reader.sc.nextInt();
                                    Reader.sc.nextLine();
                                    if (r != 1 && r != 2) System.out.println("Error. Selecciona una opción válida.");
                                    else if (r == 1){
                                        System.out.println("Eliminando...");
                                        Library.getBooksList().remove(book);
                                        System.out.println("Libro para renta eliminado");
                                        op = false;
                                    }else if(r == 2){
                                        System.out.println("Cancelando...");
                                        op = false;
                                    }
                                } while (op);
                            }
                            case 2 ->{
                                k = 1;
                                System.out.println("\t\nSelecciona el libro a eliminar:\n");
                                for (SaleBooks book : Library.getBooksSell()){
                                    System.out.printf("\n%d) ID: %s\n   Nombre: %s\n   Autor: %s",k,book.getID(),book.getTitle(),book.getAuthor());
                                    k++;
                                }
                                do {
                                    y = 0;
                                    System.out.print("\nRespuesta: ");
                                    y = Reader.sc.nextInt();
                                    Reader.sc.nextLine();
                                    y -= 1;
                                    if (y > Library.getBooksSell().size()) System.out.println("Error. Ingresa una opción válida.");
                                    else op = false;
                                } while (op);
                                SaleBooks book = Library.getBooksSell().get(y);
                                System.out.println("\nSeguro/a que deseas eliminar este libro del inventario?\n1) Sí\n2) No");
                                op = true;
                                do {
                                    r = 0;
                                    System.out.print("\nRespuesta: ");
                                    r = Reader.sc.nextInt();
                                    Reader.sc.nextLine();
                                    if (r != 1 && r != 2) System.out.println("Error. Selecciona una opción válida.");
                                    else if (r == 1){
                                        System.out.println("Eliminando...");
                                        Library.getBooksSell().remove(book);
                                        System.out.println("Libro eliminado del inventario");
                                        op = false;
                                    }else if(r == 2){
                                        System.out.println("Cancelando...");
                                        op = false;
                                    }
                                } while (op);
                            }
                            case 3 ->{
                                System.out.println("\nRegresando al menú anterior...");
                                st = false;
                            }
                            default -> System.out.println("Error. Selecciona una opción válida.");
                        }
                    }while(st);
                }
                case 6 ->{
                    System.out.println("\nRegresando al menú anterior...");
                    pt = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida.");
            }
        }while(pt);
    }
    public static void menuUsers(){
        boolean st = true, op = true;
        int opc,b,i,y,r;
        do{
            System.out.println("\nSelecciona la opción deseada:\n1) Mostrar Usuarios Registrados\n2) Usuarios con renta de libros\n3) Usuarios con historial de compras\n4) Eliminar Usuario\n5) Salir");
            System.out.print("\nOpción: ");
            opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    b=1;
                    for(Users user:Library.getUsersList()){
                        System.out.printf("\n%d) ID: %s\nNombre: %s\n   Apellidos: %s\n   Teléfono: %d\n   Edad: %d\n   Dirección: %s",b,user.getIDuser(),user.getName(),user.getLastName(),user.getCellNumber(),user.getAge(),user.getAddress());
                        System.out.println("****************");
                        b+=1;
                    }
                }
                case 2 ->{
                    b=1;
                    for(Users users:Library.getUsersList()){
                        if(!users.getRentedBooks().isEmpty()){
                            System.out.printf("\n%d) Nombre: %s\n   Apellidos: %s\n   Edad: %d\n   Libros rentados: %d",b,users.getName(),users.getLastName(),users.getAge(),users.getRentedBooks().size());
                            System.out.println("****************");
                            b+=1;
                        }
                    }
                }
                case 3 ->{
                    b=1;
                    for(Users users1:Library.getUsersList()){
                        if(!users1.getRentedBooks().isEmpty()){
                            System.out.printf("\n%d) Nombre: %s\n   Apellidos: %s\n   Edad: %d\n   Libros comprados: %d",b,users1.getName(),users1.getLastName(),users1.getAge(),users1.getSoldBooks().size());
                            System.out.println("****************");
                            b+=1;
                        }
                    }
                }
                case 4 ->{
                    i = 1;
                    System.out.println("\t\nSelecciona el usuario a eliminar:\n");
                    for (Users users2 : Library.getUsersList()){
                        System.out.printf("\n%d) Nombre: %s %s", i, users2.getName(), users2.getLastName());
                        i++;
                    }
                    do {
                        y = 0;
                        System.out.print("\nRespuesta: ");
                        y = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        y -= 1;
                        if (y > Library.getUsersList().size()) System.out.println("Error. Ingresa una opción válida.");
                        else op = false;
                    } while (op);
                    Users user = Library.getUsersList().get(y);
                    if(!user.getSoldBooks().isEmpty()) System.out.println("\nError. El usuario aún tiene libros rentados. Favor de solicitar los libros de regreso antes de eliminar");
                    else{
                        System.out.printf("\nSeguro/a que deseas eliminar a %s %s?\n1) Sí\n2) No",user.getName(),user.getLastName());
                        op = true;
                        do {
                            r = 0;
                            System.out.print("\nRespuesta: ");
                            r = Reader.sc.nextInt();
                            Reader.sc.nextLine();
                            if (r != 1 && r != 2) System.out.println("Error. Selecciona una opción válida.");
                            else if (r == 1){
                                System.out.println("Eliminando...");
                                Library.getUsersList().remove(user);
                                System.out.println("Usuario eliminado");
                                op = false;
                            }else if(r == 2){
                                System.out.println("Cancelando...");
                                op = false;
                            }
                        } while (op);
                    }
                }
                case 5 ->{
                    System.out.println("\nRegresando al menú anterior...");
                    st = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida.");
            }
        }while (st);
    }
}

File: MenuLibrary.py
import random
from Books import Books
from SaleBooks import SaleBooks
from Users import Users
from Library import Library
class MenuLibrary:
    def __init__(self):
        self.library = Library()

    def library_menu(self):
        flag = True
        while flag:
            print("\nSelecciona una opción:\n1) Registrar Libro\n2) Registrar Usuario\n3) Libros\n4) Usuarios\n5) Salir")
            x = int(input("\nOpción: "))
            try:
                if x == 1:
                    self.register_book()
                elif x == 2:
                    self.register_user()
                elif x == 3:
                    self.menu_books()
                elif x == 4:
                    self.menu_users()
                elif x == 5:
                    print("\nRegresando al menú principal...")
                    flag = False
                else:
                    print("Error. Selecciona una opción válida.")
            except ValueError:
                print("Error. Selecciona una opción válida.")

    def register_book(self):
        number = random.Random()
        flag = True
        while flag:
            print("\nSección:\n1) Renta\n2) Venta\n3) Salir")
            opc = int(input("\nOpción: "))
            try:
                if opc == 1:
                    title = input("\nIngresa el título: ")
                    author = input("\nIngresa el autor: ")
                    year = int(input("\nIngresa el año de publicación: "))
                    print("\nRegistrando libro para renta...")
                    id = number.randint(100000, 999999)
                    book = Books(id, title, author, year)
                    print("\nAñadiendo a biblioteca...")
                    self.library.add_book(book)
                    print("\nLibro registrado!")
                    flag = False
                elif opc == 2:
                    title = input("\nIngresa el título: ")
                    author = input("\nIngresa el autor: ")
                    year = int(input("\nIngresa el año de publicación: "))
                    price = float(input("\nIngresa el precio de venta: "))
                    inventory = int(input("\nIngresa el inventario: "))
                    print("\nRegistrando libro para venta...")
                    id = number.randint(100000, 999999)
                    book = SaleBooks(id, title, author, year, price, inventory)
                    book.set_available(True)
                    print("\nAñadiendo a biblioteca...")
                    self.library.add_sell_book(book)
                    print("\nLibro registrado!")
                    flag = False
                elif opc == 3:
                    print("\nRegresando al menú anterior...\n")
                    flag = False
                else:
                    print("Error. Elige una opción válida.")
            except ValueError:
                print("Error. Elige una opción válida.")

    def register_user(self):
        number = random.Random()
        sig = True
        while sig:
            print("\nConfirmar:\n1) Sí\n2) No")
            opc = int(input("\nOpción: "))
            try:
                if opc == 1:
                    name = input("\nIngresa el nombre: ")
                    last_name = input("\nIngresa los apellidos: ")
                    address = input("\nIngresa la dirección: ")
                    cell_number = int(input("\nIngresa un número de teléfono: "))
                    age = int(input("\nIngresa la edad: "))
                    if age < 18:
                        print("Error. El usuario es menor de edad.\nRegresando al menú...")
                    else:
                        print("\nRegistrando usuario..")
                        id_user = number.randint(100000, 999999)
                        user = Users(id_user, name, last_name, age, cell_number, address)
                        #rented_books = []
                        #user.set_rented_books(rented_books)
                        #sold_books = []
                        #user.set_sold_books(sold_books)
                        print("\nAñadiendo a base de datos...")
                        self.library.add_user(user)
                        print("\nUsuario registrado!")
                    sig = False
                elif opc == 2:
                    print("\nRegresando al menú anterior...\n")
                    sig = False
                else:
                    print("Error. Elige una opción válida.")
            except ValueError:
                print("Error. Elige una opción válida.")
    def menu_books(self):
        pt = True
        while pt:
            print("\nSelecciona la opción deseada:\n1) Consultar biblioteca (rentas)\n2) Consultar librería (ventas)\n3) Libros disponibles para renta\n4) Libros disponibles en venta\n5) Modificar Libro\n6) Eliminar Libro\n7) Salir")
            opc = int(input("\nOpción: "))
            if opc == 1:
                for book in self.library.get_books():
                    print("\nID: {}\nTítulo: {}\nAutor: {}\nAño de Publicación: {}".format(book.get_ID(), book.get_title(), book.get_author(), book.get_year()))
                    print("**************")
            elif opc == 2:
                for book in self.library.get_sell_books():
                    print("\nID: {}\nTítulo: {}\nAutor: {}\nAño de Publicación: {}\nPrecio: {:.2f}".format(book.get_id(), book.get_title(), book.get_author(), book.get_year(), book.get_price()))
                    print("**************")
            elif opc == 3:
                counter = 0
                for book in self.library.get_books():
                    if not book.is_rented():
                        print("\nID: {}\nTítulo: {}\nAutor: {}\nAño de Publicación: {}".format(book.get_ID(), book.get_title(), book.get_author(), book.get_year()))
                        print("**************")
                    else:
                        counter += 1
                if counter == len(self.library.get_books()):
                    print("\nNo hay libros disponibles.")
            elif opc == 4:
                counter = 0
                for book in self.library.get_sell_books():
                    if book.is_available():
                        print("\nID: {}\nTítulo: {}\nAutor: {}\nAño de Publicación: {}\nPrecio: {:.2f}\nInventario: {}".format(book.get_id(), book.get_title(), book.get_author(), book.get_year(), book.get_price(), book.get_inventory()))
                        print("**************")
                    else:
                        counter += 1
                if counter == len(self.library.get_sell_books()):
                    print("\nNo hay libros disponibles en existencia.")
            elif opc == 5:
                st = True
                while st:
                    a = int(input("\n1) Libros para Renta\n2) Libros para Venta\n3) Salir\nOpción: "))
                    if a == 1:
                        op = True
                        j = 1
                        print("\t\nSelecciona el libro a modificar:\n")
                        for i, book in enumerate(self.library.get_books(), start=1):
                            print("\n{}) ID: {}\n   Nombre: {}\n   Autor: {}".format(j, book.get_ID(), book.get_title(), book.get_author()))
                            j += 1
                        while op:
                            y = 0
                            y = int(input("\nRespuesta: "))
                            y -= 1
                            if y > len(self.library.get_books()):
                                print("Error. Ingresa una opción válida.")
                            else:
                                op = False
                        op = True
                        book = self.library.get_books()[y]
                        while op:
                            print("\t\nSelecciona el dato a modificar:\n")
                            print("\n1) Título: {}\n2) Autor: {}\n3) Año de publicación: {}\n4) Salir".format(book.get_title(), book.get_author(), book.get_year()))
                            r = int(input("\nRespuesta: "))
                            if r == 1:
                                title = input("\nIngresa el título: ")
                                print("\nModificando título...")
                                book.set_title(title)
                                print("\nTítulo modificado!")
                            elif r == 2:
                                author = input("\nIngresa el autor: ")
                                print("\nModificando autor...")
                                book.set_author(author)
                                print("\nAutor modificado!")
                            elif r == 3:
                                year = int(input("\nIngresa el año de publicación: "))
                                print("\nModificando año...")
                                book.set_year(year)
                                print("\nAño de publicación modificado!")
                            elif r == 4:
                                print("\nRegresando al menú anterior...")
                                op = False
                            else:
                                print("Error. Selecciona una opción válida")
                    elif a == 2:
                        op = True
                        k = 1
                        print("\t\nSelecciona el libro a modificar:\n")
                        for i, book in enumerate(self.library.get_sell_books(), start=1):
                            print("\n{}) ID: {}\n   Título: {}\n   Autor: {}".format(k, book.get_id(), book.get_title(), book.get_author()))
                            k += 1
                        while op:
                            y = 0
                            y = int(input("\nRespuesta: "))
                            y -= 1
                            if y > len(self.library.get_sell_books()):
                                print("Error. Ingresa una opción válida.")
                            else:
                                op = False
                        op = True
                        book = self.library.get_sell_books()[y]
                        while op:
                            print("\t\nSelecciona el dato a modificar:\n")
                            print("\n1) Título: {}\n2) Autor: {}\n3) Año de publicación: {}\n4) Precio de venta: {:.2f}\n5) Inventario: {}\n6) Salir".format(book.get_title(), book.get_author(), book.get_year(), book.get_price(), book.get_inventory()))
                            r = int(input("\nRespuesta: "))
                            if r == 1:
                                title = input("\nIngresa el título: ")
                                print("\nModificando título...")
                                book.set_title(title)
                                print("\nTítulo modificado!")
                            elif r == 2:
                                author = input("\nIngresa el autor: ")
                                print("\nModificando autor...")
                                book.set_author(author)
                                print("\nAutor modificado!")
                            elif r == 3:
                                year = int(input("\nIngresa el año de publicación: "))
                                print("\nModificando año...")
                                book.set_year(year)
                                print("\nAño de publicación modificado!")
                            elif r == 4:
                                price = float(input("\nIngresa el precio de venta: "))
                                print("\nModificando precio...")
                                book.set_price(price)
                                print("\nPrecio de venta modificado!")
                            elif r == 5:
                                inventory = int(input("\nIngresa el nuevo inventario: "))
                                print("\nActualizando inventario...")
                                book.set_inventory(inventory)
                                if book.get_inventory() == 0:
                                    book.set_available(False)
                                elif book.get_inventory() > 0:
                                    book.set_available(True)
                                print("\nInventario actualizado!")
                            elif r == 6:
                                print("\nRegresando al menú anterior...")
                                op = False
                            else:
                                print("Error. Selecciona una opción válida")
                    elif a == 3:
                        print("\nRegresando al menú anterior...")
                        st = False
                    else:
                        print("Error. Selecciona una opción válida.")
            elif opc == 6:
                st = True
                while st:
                    a = int(input("\n1) Libros para Renta\n2) Libros para Venta\n3) Salir\nOpción: "))
                    if a == 1:
                        op = True
                        j = 1
                        print("\t\nSelecciona el libro a eliminar:\n")
                        for i, book in enumerate(self.library.get_books(), start=1):
                            print("\n{}) ID: {}\n   Título: {}\n   Autor: {}".format(j, book.get_ID(), book.get_title(), book.get_author()))
                            j += 1
                        while op:
                            y = 0
                            y = int(input("\nRespuesta: "))
                            y -= 1
                            if y > len(self.library.get_books()):
                                print("Error. Ingresa una opción válida.")
                            else:
                                op = False
                        book = self.library.get_books()[y]
                        if book.is_rented():
                            print("Error. El libro fue rentado por un usuario. Favor de retornar el libro antes de borrarlo.")
                        else:
                            print("\nSeguro/a que deseas eliminar este libro?\n1) Sí\n2) No")
                            op = True
                            while op:
                                r = 0
                                r = int(input("\nRespuesta: "))
                                if r != 1 and r != 2:
                                    print("Error. Selecciona una opción válida.")
                                elif r == 1:
                                    print("Eliminando...")
                                    self.library.get_books().remove(book)
                                    print("Libro para renta eliminado")
                                    op = False
                                elif r == 2:
                                    print("Cancelando...")
                                    op = False
                    elif a == 2:
                        op = True
                        k = 1
                        print("\t\nSelecciona el libro a eliminar:\n")
                        for i, book in enumerate(self.library.get_sell_books(), start=1):
                            print("\n{}) ID: {}\n   Título: {}\n   Autor: {}".format(k, book.get_id(), book.get_title(), book.get_author()))
                            k += 1
                        while op:
                            y = 0
                            y = int(input("\nRespuesta: "))
                            y -= 1
                            if y > len(self.library.get_sell_books()):
                                print("Error. Ingresa una opción válida.")
                            else:
                                op = False
                        book = self.library.get_sell_books()[y]
                        print("\nSeguro/a que deseas eliminar este libro del inventario?\n1) Sí\n2) No")
                        op = True
                        while op:
                            r = 0
                            r = int(input("\nRespuesta: "))
                            if r != 1 and r != 2:
                                print("Error. Selecciona una opción válida.")
                            elif r == 1:
                                print("Eliminando...")
                                self.library.get_sell_books().remove(book)
                                print("Libro eliminado del inventario")
                                op = False
                            elif r == 2:
                                print("Cancelando...")
                                op = False
                    elif a == 3:
                        print("\nRegresando al menú anterior...")
                        st = False
                    else:
                        print("Error. Selecciona una opción válida.")
            elif opc == 7:
                print("\nRegresando al menú anterior...")
                pt = False
            else:
                print("Error. Selecciona una opción válida.")
    def menu_users(self):
        st = True
        while st:
            print("\nSelecciona la opción deseada:\n1) Mostrar Usuarios Registrados\n2) Usuarios con renta de libros\n3) Usuarios con historial de compras\n4) Modificar Usuario\n5) Eliminar Usuario\n6) Salir")
            opc = int(input("\nOpción: "))
            if opc == 1:
                b = 1
                for user in self.library.get_users():
                    print("\n{}) ID: {}\n   Nombre: {}\n   Apellidos: {}\n   Teléfono: {}\n   Edad: {}\n   Dirección: {}".format(b, user.get_IDuser(), user.get_name(), user.get_lastName(), user.get_cellNumber(), user.get_age(), user.get_address()))
                    print("****************")
                    b += 1
            elif opc == 2:
                b = 1
                counter = 0
                for user in self.library.get_users():
                    if user.get_rentedBooks():
                        print("\n{}) Nombre: {}\n   Apellidos: {}\n   Edad: {}\n   Libros rentados: {}".format(b, user.get_name(), user.get_lastName(), user.get_age(), len(user.get_rentedBooks())))
                        print("****************")
                        b += 1
                    else:
                        counter += 1
                        if counter == len(self.library.get_users()):
                            print("\nNo hay usuarios con rentas existentes")
            elif opc == 3:
                b = 1
                counter = 0
                for user in self.library.get_users():
                    if user.get_soldBooks():
                        print("\n{}) Nombre: {}\n   Apellidos: {}\n   Edad: {}\n   Libros comprados: {}".format(b, user.get_name(), user.get_lastName(), user.get_age(), len(user.get_soldBooks())))
                        print("****************")
                        b += 1
                    else:
                        counter += 1
                        if counter == len(self.library.get_users()):
                            print("\nNo existen usuarios con compras efectuadas")
            elif opc == 4:
                op = True
                i = 1
                print("\t\nSelecciona el usuario a modificar:\n")
                for user in self.library.get_users():
                    print("\n{}) Nombre: {} {}".format(i, user.get_name(), user.get_lastName()))
                    i += 1
                while op:
                    y = 0
                    y = int(input("\nRespuesta: ")) - 1
                    if y > len(self.library.get_users()):
                        print("Error. Ingresa una opción válida.")
                    else:
                        op = False
                op = True
                user = self.library.get_users()[y]
                while op:
                    print("\t\nSelecciona el dato a modificar:\n")
                    print("\n1) Nombre: {}\n2) Apellidos: {}\n3) Teléfono: {}\n4) Edad: {}\n5) Dirección: {}\n6) Salir".format(user.get_name(), user.get_lastName(), user.get_cellNumber(), user.get_age(), user.get_address()))
                    r = int(input("\nRespuesta: "))
                    if r == 1:
                        name = input("\nIngresa el nombre: ")
                        print("\nModificando nombre...")
                        user.set_name(name)
                        print("\nNombre modificado!")
                    elif r == 2:
                        last_name = input("\nIngresa el apellido: ")
                        print("\nModificando apellido...")
                        user.set_lastName(last_name)
                        print("\nApellido modificado!")
                    elif r == 3:
                        cell_number = int(input("\nIngresa el número: "))
                        print("\nModificando teléfono...")
                        user.set_cellNumber(cell_number)
                        print("\nTeléfono modificado!")
                    elif r == 4:
                        ag = True
                        while ag:
                            age = int(input("\nIngresa la edad: "))
                            if age < 18:
                                print("\nError. El usuario no puede ser menor de edad")
                            else:
                                print("\nModificando edad...")
                                user.set_age(age)
                                print("\nEdad modificada!")
                                ag = False
                    elif r == 5:
                        address = input("\nIngresa la dirección: ")
                        print("\nModificando dirección...")
                        user.set_address(address)
                        print("\nDirección modificada!")
                    elif r == 6:
                        print("\nRegresando al menú principal...")
                        op = False
                    else:
                        print("Error. Selecciona una opción válida")
            elif opc == 5:
                op = True
                i = 1
                print("\t\nSelecciona el usuario a eliminar:\n")
                for user in self.library.get_users():
                    print("\n{}) Nombre: {} {}".format(i, user.get_name(), user.get_lastName()))
                    i += 1
                while op:
                    y = 0
                    y = int(input("\nRespuesta: ")) - 1
                    if y > len(self.library.get_users()):
                        print("Error. Ingresa una opción válida.")
                    else:
                        op = False
                user = self.library.get_users()[y]
                if not user.get_rentedBooks():
                    print("\nSeguro/a que deseas eliminar a {} {}?\n1) Sí\n2) No".format(user.get_name(), user.get_lastName()))
                    op = True
                    while op:
                        r = int(input("\nRespuesta: "))
                        if r != 1 and r != 2:
                            print("Error. Selecciona una opción válida.")
                        elif r == 1:
                            print("Eliminando...")
                            self.library.get_users().remove(user)
                            print("Usuario eliminado")
                            op = False
                        else:
                            print("Cancelando...")
                            op = False
                else:
                    print("\nError. El usuario aún tiene libros rentados. Favor de solicitar los libros de regreso antes de eliminar")
            elif opc == 6:
                print("\nRegresando al menú anterior...")
                st = False
            else:
                print("Error. Selecciona una opción válida.")